### PR TITLE
VACMS 19842 - Situation Update Banners to use Vets-API and depend on feature toggle

### PR DESCRIPTION
**Pull request by eselkin**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No

## Summary

- Updated Situation Banner component to use vets-api resource /v0/banners, broke out helper function for possible stubbing, added tests.
- Sitewide team
- Uses CMS flipper for whether to use data-widget-type on content-build and vets-api flipper `banner_use_alternative_banners`
 
## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-cms/issues/19842

## Testing done

- Manual testing (Banners::UpdateAll run on local vets-api from staging CMS, tested several types of banners). Tested with toggle on vets-api on and off. Tested building with and without CMS feature toggle on. Tested on health-care sub pages (Facilities and others) and on va-manila-clinic pages.
- Added unit tests to cover 94.56% of lines

## Screenshots

<details>
<summary>Birmingham - Info Banner - not dismissible - desktop</summary>
Screenshot 2024-12-05 at 8 53 29 PM
Screenshot 2024-12-05 at 8 54 11 PM
</details>

<details>
<summary>Birmingham - Info Banner - not dismissible - 481px</summary>
Screenshot 2024-12-05 at 8 54 42 PM
</details>

<details>
<summary>Maine - Info Banner - dismissible - desktop </summary>
Screenshot 2024-12-05 at 8 53 38 PM
Screenshot 2024-12-05 at 8 53 58 PM
</details>

<details>
<summary>Maine - Info Banner - dismissible - 481px</summary>
Screenshot 2024-12-05 at 8 54 54 PM
</details>

<details>
<summary>Birmingham - Info Banner - operating status page</summary>
Screenshot 2024-12-05 at 9 20 50 PM
</details>

<details>
<summary></summary>
</details>

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/static-pages/situation-updates-banner/bannerContainer.jsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  );
+  const isLoadingFeatureFlags = useToggleLoadingValue();
+
+  const [bannerState, setBannerState] = useState(null);
+
+  useEffect(() => {
+    let pathForSituationUpdate = window.location.pathname;
+    if (pathForSituationUpdate.endsWith('/')) {
+      pathForSituationUpdate = pathForSituationUpdate.slice(0, -1);
+    }
```
Comment (SnowboardTechie) [inline src/applications/static-pages/situation-updates-banner/bannerContainer.jsx:24]: Bit of a nit so feel free to ignore, but I think this looks better using replace regex
```suggestion
     pathForSituationUpdate = pathForSituationUpdate.replace(/\/$/, '');
```

**Review by SnowboardTechie — DISMISSED** [review]
This looks great! 👍🏻

**Review by SnowboardTechie — DISMISSED** [review]
👍🏻

File: src/applications/static-pages/situation-updates-banner/bannerContainer.jsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  const alternativeBannersEnabled = useToggleValue(
+    TOGGLE_NAMES.bannerUseAlternativeBanners,
+  );
+  const isLoadingFeatureFlags = useToggleLoadingValue();
+
+  const [bannerState, setBannerState] = useState(null);
+
+  useEffect(() => {
+    let pathForSituationUpdate = window.location.pathname;
+     pathForSituationUpdate = pathForSituationUpdate.replace(/\/$/, '');
```
Comment (SnowboardTechie) [inline src/applications/static-pages/situation-updates-banner/bannerContainer.jsx:22]:
```suggestion
    pathForSituationUpdate = pathForSituationUpdate.replace(/\/$/, '');
```

**Review by SnowboardTechie — DISMISSED** [review]
👍🏻 👍🏻 👍🏻

**Review by SnowboardTechie — APPROVED** [review]
Awesome stuff 👍🏻 👍🏻 👍🏻 👍🏻

File: src/applications/static-pages/situation-updates-banner/situationUpdateBanner.jsx [hunk 3]
```diff
@@ -10,12 +12,19 @@
...
       headline={headline}
       show-close={showClose}
+      disable-analytics
+      onclick={recordEvent({
+        event: 'nav-warning-alert-box-content-link-click',
+        alertBoxHeading: '{{ entity.title }}',
+      })}
     >
-      <p>{content}</p>
+      {/* eslint-disable-next-line react/no-danger */}
```
[the remark this answers is not in the document]
Comment (eselkin) [reply src/applications/static-pages/situation-updates-banner/situationUpdateBanner.jsx:23]: discussed

File: src/applications/static-pages/situation-updates-banner/tests/bannerContainer.unit.spec.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+import { mockApiRequest } from 'platform/testing/unit/helpers';
+
+import { BannerContainer } from '../bannerContainer';
+import { birmingham, maine } from './mocks/mockSituationUpdatesBanner';
+
+const getData = ({ isLoading = false, useBanners = true } = {}) => ({
+  featureToggles: {
+    loading: isLoading,
+    bannerUseAlternativeBanners: useBanners,
+    // eslint-disable-next-line camelcase
```
[the remark this answers is not in the document]
Comment (eselkin) [reply src/applications/static-pages/situation-updates-banner/tests/bannerContainer.unit.spec.js:22]: object does not obey the eslint rules

**Pull request merged**